scripts/workspace-app-reg.py

```diff
@@ -0,0 +1,194 @@
+
+    if cont.lower() != "y" and cont.lower() != "yes":
+        sys.exit(0)
+
+
+def ensure_sp(appId: str):
+    sps = graph.get(f"/servicePrincipals?$filter=appid eq '{appId}'")
+    if len(sps) == 0:
+        graph.post("/servicePrincipals", json={
+            "appId": appId,
+            "appRoleAssignmentRequired": True,
+            "tags": ['WindowsAzureActiveDirectoryIntegratedApp'],
+        }, headers={'Content-Type': 'application/json'})
+
+
+def main():
+    global me
+    me = graph.get("/me")
+
+    if not options.force:
+        _double_check()
+
+    existing_app = get_existing_app(options.app_name)
+
+    # Generate new Guids if needed
+    ownerRoleId = str(uuid.uuid4())
+    researcherRoleId = str(uuid.uuid4())
+
+    # Get existing role and scope ids (in case we are updating the existing app)
+    if existing_app is not None:
+        ownerRoleId = [r['id'] for r in existing_app['appRoles'] if r['value'] == 'WorkspaceOwner'][0]
+        researcherRoleId = [r['id'] for r in existing_app['appRoles'] if r['value'] == 'WorkspaceResearcher'][0]
+
+    appRoles = [
+        {
+            "id": researcherRoleId,
+            "allowedMemberTypes": ["User"],
+            "description": f"Provides access to the {options.tre_name} workspace {options.workspace_name}.",
+            "displayName": "Researchers",
+            "isEnabled": True,
+            "origin": "Application",
+            "value": "WorkspaceResearcher"
+        },
+        {
+            "id": ownerRoleId,
+            "allowedMemberTypes": ["User"],
+            "description": f"Provides ownership access to the {options.tre_name} workspace {options.workspace_name}.",
+            "displayName": "Owners",
+            "isEnabled": True,
+            "origin": "Application",
+            "value": "WorkspaceOwner"
+        }
+    ]
+
+    # Define the API application
+    apiApp = {
+        "displayName": options.app_name,
+        "appRoles": appRoles,
+        "signInAudience": "AzureADMyOrg"
+    }
+
+    if existing_app is not None:
+        apiAppId = existing_app['appId']
+        # Update
+        resp = graph.patch(f"/applications/{existing_app['id']}", json=apiApp, headers={'Content-Type': 'application/json'})
+        if resp.ok:
+            print(f"Updated application \"{existing_app['displayName']}\" (appid={apiAppId})")
+        else:
+            content = resp.json()
+            print(content)
+    else:
+        # Create
+        resp = graph.post("/applications", json=apiApp, headers={'Content-Type': 'application/json'})
+        content = resp.json()
+        if resp.ok:
+            apiAppId = content['appId']
+            print(f"Created application \"{content['displayName']}\" (appid={apiAppId})")
+            graph.post(f"/applications/{content['id']}/owners/$ref")
+        else:
+            print(content)
+
+    if apiAppId is not None and len(apiAppId) != 0:
```

Comment:
if not apiAppId

Reply:
This line has gone now, but useful to know the simpler form.